src/main/java/pl/psnc/dei/ui/components/ListImportComponent.java

```diff
@@ -0,0 +1,237 @@
+package pl.psnc.dei.ui.components;
+
+import com.vaadin.flow.component.Component;
+import com.vaadin.flow.component.button.Button;
+import com.vaadin.flow.component.dependency.StyleSheet;
+import com.vaadin.flow.component.html.Label;
+import com.vaadin.flow.component.icon.Icon;
+import com.vaadin.flow.component.icon.VaadinIcon;
+import com.vaadin.flow.component.orderedlayout.FlexComponent;
+import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
+import com.vaadin.flow.component.orderedlayout.VerticalLayout;
+import com.vaadin.flow.component.select.Select;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import pl.psnc.dei.model.DAO.ImportsRepository;
+import pl.psnc.dei.model.Import;
+import pl.psnc.dei.ui.components.imports.ImportResult;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@StyleSheet("frontend://styles/styles.css")
+public class ListImportComponent extends VerticalLayout implements ResultComponent {
+	public static final int DEFAULT_PAGE_SIZE = 3;
+
+	private static final String NAME = "Name";
+
+	private static final String CREATION_DATE = "Creation date";
+
+	private static final String STATUS = "Status";
+
+	private static final String ACTION = "Action";
+
+	// navigation bar with results count and page navigation
+	private HorizontalLayout navigationBar;
+
+	private PageNavigationComponent pageNavigationComponent;
+
+	private VerticalLayout importList;
```

Comment:
Generally I think that if not necessary we shouldn't implement this list from scratch. I would try to use https://vaadin.com/components/vaadin-grid or something like this. There are many nice free components from Vaadin that could be used.

Reply:
Additonally this grid allows sorting, has all the alignments correct and so on...